helix-core/src/main/java/org/apache/helix/monitoring/metrics/model/CountMetric.java

```diff
@@ -19,23 +19,46 @@
  * under the License.
  */
 
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMetric;
 import org.apache.helix.monitoring.mbeans.dynamicMBeans.SimpleDynamicMetric;
 
 /**
  * Represents a count metric and defines methods to help with calculation. A count metric gives a
  * gauge value of a certain property.
  */
-public abstract class CountMetric<V> extends SimpleDynamicMetric<V> implements Metric {
-  protected V _count;
+public abstract class CountMetric extends SimpleDynamicMetric<Long> implements Metric {
 
   /**
-   * Instantiates a new Simple dynamic metric.
+   * Instantiates a new count metric.
+   *
    * @param metricName the metric name
-   * @param metricObject the metric object
+   * @param initCount the initial count
    */
-  public CountMetric(String metricName, V metricObject) {
-    super(metricName, metricObject);
+  public CountMetric(String metricName, long initCount) {
```

Comment:
I recall having this discussion with you. Should we specify the exact type in the subclass of `CountMetric` in their constructors? That allows other subclasses of CountMetric to specify their own SimpleDynamicMetric.

Reply:
Same reason, it won't workout if getLastEmittedMetricValue() is returning long.
And for the purpose of counting, long should be good enough.